### PR TITLE
Return 204 not 500 on filtered pipeline

### PR DESCRIPTION
**Pull request by 6543**
the error check always failed, as the custom errors did not have the `Is(error) bool` interface implemented

backport bugfix part of  #2216

### CODE REVIEW

File: server/pipeline/errors.go [hunk 1]
```diff
@@ -18,22 +18,37 @@ type ErrNotFound struct {
 	Msg string
 }
 
-func (e ErrNotFound) Error() string {
+func (e *ErrNotFound) Error() string {
```
Comment (lafriks) [inline server/pipeline/errors.go:21]:
```suggestion
func (e ErrNotFound) Error() string {
```

File: server/pipeline/errors.go [hunk 2]
```diff
@@ -18,22 +18,37 @@ type ErrNotFound struct {
...
 	return e.Msg
 }
 
+func (e *ErrNotFound) Is(target error) bool {
+	_, ok := target.(*ErrNotFound) //nolint:errorlint
+	return ok
+}
```
Comment (lafriks) [inline server/pipeline/errors.go:28]:
```suggestion
func (e ErrNotFound) Is(target error) bool {
	_, ok := target.(ErrNotFound)
	if !ok {
		_, ok = target.(*ErrNotFound)
	}
	return ok
}
```

File: server/pipeline/errors.go (hunk 1 above)
Comment (lafriks) [reply server/pipeline/errors.go:21]: this way it will work for both non-pointer and pointer variables

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
I had noticed this bug for quite some time but had no time to look into it so thanks for fixing it ❤️

**Comment by 6543** [comment]
applied the changes here to the related pull for main too ...

**Pull request merged**